src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -1291,3 +1314,56 @@ bool InputStateMachineEngine::_GetSGRXYPosition(const std::basic_string_view<siz
 
     return true;
 }
+
+// Method Description:
+// - Attempt to parse our parameters into a win32-input-mode serialized KeyEvent.
+// Arguments:
+// - parameters: the list of numbers to parse into values for the KeyEvent.
+// - key: receives the values of the deserialized KeyEvent.
+// Return Value:
+// - true if we successfully parsed the key event.
+bool InputStateMachineEngine::_GenerateWin32Key(const std::basic_string_view<size_t> parameters,
+                                                KeyEvent& key)
+{
+    // Sequences are formatted as follows:
+    //
+    // ^[ [ Vk ; Sc ; Uc ; Kd ; Cs ; Rc _
+    //
+    //      Vk: the value of wVirtualKeyCode - any number. If omitted, defaults to '0'.
+    //      Sc: the value of wVirtualScanCode - any number. If omitted, defaults to '0'.
+    //      Uc: the decimal value of UnicodeChar - for example, NUL is "0", LF is
+    //          "10", the character 'A' is "65". If omitted, defaults to '0'.
+    //      Kd: the value of bKeyDown - either a '0' or '1'. If omitted, defaults to '0'.
+    //      Cs: the value of dwControlKeyState - any number. If omitted, defaults to '0'.
+    //      Rc: the value of wRepeatCount - any number. If omitted, defaults to '0'.
+
+    if (parameters.size() > 6)

```

Comment:
Do we have to check for...
- parameters.size() == 0 (or empty)
- parameters == nullptr